sockshop/catalogue/middlewares.go

```diff
@@ -25,7 +25,7 @@ type loggingMiddleware struct {
 	logger log.Logger
 }
 
-func (mw loggingMiddleware) List(tags []string, order string, pageNum, pageSize int) (socks []Sock) {
+func (mw loggingMiddleware) List(tags []string, order string, pageNum, pageSize int) (socks []Sock, err error) {
```

Comment:
Totally up to you, but it's nice to include the error in the deferred Log call. Just add a line,

```go
   . . .
    "result", len(socks),
    "err", err, // <--
    "took", time.Since(begin),
   . . .
```